drivers/scheduler/anthos/anthos.go

```diff
@@ -978,6 +978,38 @@ func getStoragePDBMinAvailableSet() (int, error) {
 	return pdbVal, nil
 }
 
+// DeleteNode delete nodes from a cluster and return error if fails
+func (anth *anthos) DeleteNode(node node.Node) error {
+	if err := anth.getUserClusterName(); err != nil {
+		return err
+	}
+	if err := anth.Ops.DeleteMachine(context.TODO(), node.Name); err != nil {
```

Comment:
Should there be a retry loop here to wait till the machine gets deleted?

Reply:
Incorporated !!